### PR TITLE
oci: add oci cleanup

### PR DESCRIPTION
**Pull request by croissanne**

### CODE REVIEW

**Review by ondrejbudai — COMMENTED** [review]
Looks good! You also need to add it here: https://github.com/osbuild/cloud-cleaner/blob/main/.github/workflows/run_ib.yml

File: oci.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    INSTANCE_DATA=$("$OCI_CMD" compute instance get --instance-id "$id" | jq -r ".data")
+
+    if [[ $(echo "$INSTANCE_DATA" | jq -r '.["freeform-tags"].persist') = true ]]; then
+        echo "Instance is tagged with persist=true"
+        continue
+    fi
+
+    TIME_CREATED=$(echo "$INSTANCE_DATA" | jq -r '.["time-created"]')
+    if [[ $(date -d "$TIME_CREATED" +%s) -gt "$DELETE_TIME" ]]; then
+        echo "Instance was created less than 6 hours ago"
```
Comment (jrusz) [inline oci.sh:41]:
```suggestion
        echo "Instance was created less than $DELETE_TIME hours ago"
```

File: oci.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+$OCI_CMD setup repair-file-permissions --file "${TEMPDIR}/priv_key.pem"
+$OCI_CMD setup repair-file-permissions --file "$OCI_CONFIG"
+
+greenprint "🧹 cleaning up instances"
+INSTANCES=$("$OCI_CMD" compute instance list -c "$OCI_COMPARTMENT" | jq -r ".data[].id")
+for i in "$INSTANCES"; do
+    INSTANCE_DATA=$("$OCI_CMD" compute instance get --instance-id "$id" | jq -r ".data")
+
+    if [[ $(echo "$INSTANCE_DATA" | jq -r '.["freeform-tags"].persist') = true ]]; then
+        echo "Instance is tagged with persist=true"
```
Comment (jrusz) [inline oci.sh:35]:
```suggestion
        echo "Instance $i is tagged with persist=true"
```

File: oci.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    "$OCI_CMD" compute instance terminate --force --instance-id "$i"
+done
+
+greenprint "🧹 cleaning up images"
+IMAGES=$("$OCI_CMD" compute image list -c "$OCI_COMPARTMENT" --all | jq -r ".data[] | select(.[\"compartment-id\"] == \"$OCI_COMPARTMENT\").id")
+for i in "$IMAGES"; do
+    IMAGE_DATA=$("$OCI_CMD" compute image get --image-id "$i" | jq -r ".data")
+
+    if [[ $(echo "$IMAGE_DATA" | jq -r '.["freeform-tags"].persist') = true ]]; then
+        echo "Image is tagged with persist=true"
```
Comment (jrusz) [inline oci.sh:55]:
```suggestion
        echo "Image $i is tagged with persist=true"
```

File: oci.sh [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        continue
+    fi
+
+    TIME_CREATED=$(echo "$IMAGE_DATA" | jq -r '.["time-created"]')
+    if [[ $(date -d "$TIME_CREATED" +%s) -gt "$DELETE_TIME" ]]; then
+        echo "IMAGE was created less than 6 hours ago"
```
Comment (jrusz) [inline oci.sh:61]:
```suggestion
        echo "IMAGE $i was created less than $DELETE_TIME hours ago"
```

**Review by jrusz — COMMENTED** [review]
Looks good, I just added some suggestions to improve the `echo` outputs with more info. Did you try some dry-run or something to check if it works? I am not familiar with this env at all...

**Comment by jrusz** [comment]
I guess it should also be added here https://github.com/osbuild/cloud-cleaner/blob/main/cloud_cleaner.sh and since I mentioned dry run I realized that this is not supporting dry run like aws/aws/gcp do, although vmware doesn't seem to have it... Do you want to implement it here?

**Review by jrusz — APPROVED** [review]

**Pull request merged**